pkg/securitycontext/util.go

```diff
@@ -89,6 +89,27 @@ func HasRootRunAsUser(container *v1.Container) bool {
 	return HasRunAsUser(container) && HasRootUID(container)
 }
 
+// HasNonRootGID returns true if the runAsGroup is set and is greater than 0.
+func HasRootGID(container *v1.Container) bool {
+	if container.SecurityContext == nil {
+		return false
+	}
+	if container.SecurityContext.RunAsGroup == nil {
+		return false
+	}
+	return *container.SecurityContext.RunAsGroup == 0
+}
+
+// HasRunAsGroup determines if the sc's runAsGroup field is set.
+func HasRunAsGroup(container *v1.Container) bool {
+	return container.SecurityContext != nil && container.SecurityContext.RunAsGroup != nil
+}
+
+// HasRootRunAsGroup returns true if the run as group is set and it is set to 0.
+func HasRootRunAsGroup(container *v1.Container) bool {
+	return HasRunAsGroup(container) && HasRootGID(container)
```

Comment:
I don't see where we're using these 3 functions. Why do we need them?

Reply:
i added them since there were parallels for HasRootRunAsUser, i will remove if they arent being used